controllers/dependencies.go

```diff
@@ -0,0 +1,106 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"os"
+
+	"path/filepath"
+
+	git "github.com/go-git/go-git/v5"
+	tekton "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	OPERATOR_PIPELINES_REPO = "https://github.com/redhat-openshift-ecosystem/operator-pipelines.git"
+	REPO_CLONE_PATH         = "/tmp/operator-pipelines/"
+	PIPELINE_MANIFESTS_PATH = "ansible/roles/operator-pipeline/templates/openshift/pipelines"
+	TASK_MANIFESTS_PATH     = "ansible/roles/operator-pipeline/templates/openshift/tasks"
+)
+
+func (r *OperatorPipelineReconciler) reconcilePipelineDependencies(meta metav1.ObjectMeta) error {
+
+	// Cloning operator-pipelines project to retrieve pipelines and tasks
+	// yaml manifests that need to be applied beforehand
+	// ref: https://github.com/redhat-openshift-ecosystem/certification-releases/blob/main/4.9/ga/ci-pipeline.md#step-6---install-the-certification-pipeline-and-dependencies-into-the-cluster
+
+	_, err := git.PlainClone(REPO_CLONE_PATH, false, &git.CloneOptions{
+		URL: OPERATOR_PIPELINES_REPO,
+		// Progress: os.Stdout,
+	})
+	if err != nil {
+		log.Log.Info("Couldn't clone the repository for operator-pipelines: " + err.Error())
+		return err
+	}
+	defer r.removePipelineDependencyFiles(REPO_CLONE_PATH)
+
+	paths := []string{PIPELINE_MANIFESTS_PATH, TASK_MANIFESTS_PATH}
+	for _, path := range paths {
+
+		// base repository root + specific yaml manifest directory (pipelines or tasks)
+		root := REPO_CLONE_PATH + path
+
+		// walking through the each directory (pipelines or tasks)
+		err = filepath.Walk(root, func(filePath string, info os.FileInfo, err error) error {
+
+			// For each file NOT directories
+			if !info.IsDir() {
+				var pipeline tekton.Pipeline
+				var task tekton.Task
+				fmt.Print(path)
```

Comment:
this seems to be a leftover from the debugging. Could be removed.

Reply:
That's also done.